### ISSUE TITLE
Tie breaker strategy

### ISSUE THREAD

**Issue by strinsberg** [SibylLab/Program-Wars#583]
**Is your feature request related to a problem? Please describe.**
I did not get a tie breaking strategy back into the game after the re-write. Currently the first player in the list will win if there is a tie, which may be ok for games with only one human player. This way if you tie the bot you just win. But with 2 human players it would be frustrating. It is also much more likely to happen if we introduce the 4 player game types over time.

**Describe the solution you'd like**
A collection of bonus objective comparisons would probably be the best way to resolve this. Like in sports if two teams have the same points in a season they look at total wins, goals, goal differential until they find a category that is not tied. Ie) use bonus objective scores that we feel should be the highest priority for the player, or reward good practice.

**Comment by maimoona-bashir** [comment]
**Tie breaker:**

check number of complete stacks (complete program)

Compare number of complete stacks (CS), player having maximum numbers of CS will win the game.

**Complete stack:**
{method card
instruction card
repeat 
variable}

**Comment by johnanvik** [comment]
@maimoona-bashir, I think this was resolved? Please confirm.

**Comment by maimoona-bashir** [comment]
yes, it was. we discussed about it.

**Comment by johnanvik** [comment]
@maimoona-bashir, can you summarize what the end decision was from the discussion here for future reference if we need it.

**Comment by maimoona-bashir** [comment]
Actually @strinsberg suggested that would be too much for version 2, we could do this in version 3 by comparing bonus scores and prioritising them.
